### PR TITLE
feat: add doc search to 404 page

### PR DESCRIPTION
**Pull request by vannyle**
This pull request brings the docs search to the 404 page if the current URL of 404 page includes `/docs/`.

**TODO**
- [x] add loader for CTA component

### CODE REVIEW

**Comment by bolotskydev** [comment]
LGTM, but I have problems with slow search bar appearing, let's switch from JS-based conditional rendering to CSS ? 🤔

File: src/components/shared/search/input/input.jsx [hunk 1]
```diff
@@ -5,25 +5,36 @@ import { connectSearchBox } from 'react-instantsearch-dom';
...
+          'search-input w-full appearance-none bg-white text-black placeholder-gray-3 outline-none',
+          isNotFoundPage
+            ? 'h-16 rounded-[110px] border-2 border-gray-2 pl-14 pr-6 text-xl md:text-lg xs:pr-2.5 xs:pl-9 xs:text-base'
+            : 'h-9 rounded border border-gray-3 pl-9 pr-2.5 ',
+          isNotFoundPage && hasFocus && currentRefinement && 'rounded-[29px]',
+          hasFocus && currentRefinement && 'rounded-b-none border-b'
+        )}
+        type="search"
+        value={currentRefinement}
+        placeholder={isNotFoundPage ? 'Search from other pages' : 'Search'}
```
Comment (danieltprice) [inline src/components/shared/search/input/input.jsx:29]: Should this be search *for* other pages?

File: src/components/shared/search/input/input.jsx [hunk 2]
```diff
@@ -5,25 +5,36 @@ import { connectSearchBox } from 'react-instantsearch-dom';
...
+        autoComplete="off"
+        aria-label={isNotFoundPage ? 'Search from other pages' : 'Search'}
```
Comment (danieltprice) [inline src/components/shared/search/input/input.jsx:31]: Should this be search for other pages?

**Review by danieltprice — CHANGES_REQUESTED** [review]
A couple of comments regarding "Search from" vs "Search for" in the Search box.

File: src/components/shared/search/input/input.jsx (hunk 2 above)
Comment (vannyle) [reply src/components/shared/search/input/input.jsx:31]: It makes sense, I'll make the change

**Review by bolotskydev — APPROVED** [review]

**Comment by bolotskydev** [comment]
@danieltprice we are done with this, please, review one last time 🙏

**Review by danieltprice — APPROVED** [review]
LGTM! Thanks for the changes!

**Pull request merged**